### DISCUSSION TITLE
How to best upgrade from mcguirepr89 to Nachtzuster's fork?

### DISCUSSION THREAD

**Discussion by slnc** [Nachtzuster/BirdNET-Pi#246] [Q&A] [answered]
Hi, I'm currently running the latest mcguirepr89 BirdNET-Pi version and I just discovered this fork. Can I just change the git remote for my installation to point to this repo and trigger a system update through the UI to migrate or am I better off doing a backup followed by a clean install?

Thanks!

**Comment by blogdemoi** [comment]
Hi, as for me I did an upgrade from mcguirepr89 and everything went well. However, do a backup, do an image of your sd card e.g with dd if using Linux or another software.

**Reply by slnc** [reply]
> Hi, as for me I did an upgrade from mcguirepr89 and everything went well. However, do a backup, do an image of your sd card e.g with dd if using Linux or another software.

Thanks. You upgraded just by changing the git remote on your pi and then triggering an update via the ui?

**Reply by blogdemoi** [reply, marked as answer]
> Migrating
> 
> Before switching, make sure your installation is fully up-to-date. Also make sure to have a backup, that is also the only way to get back to the original BirdNET-Pi. Please note that upgrading your underlying OS to Bookworm is not going to work. Please stick to Bullseye. If you do want Bookworm, you need to start from a fresh install and copy back your data. (remember the backup!)
> 
> Run these commands to migrate to this repo:
> 
> git remote remove origin
> git remote add origin https://github.com/Nachtzuster/BirdNET-Pi.git
> ./scripts/update_birdnet.sh
>

**Reply by slnc** [reply]
I've migrated successfully to this fork but there was no `./scripts/backup_data.sh` on mcquirepr89's fork (762732c13434c5ab81d667da952e2345a04519cb) and I missed the backup option available through the UI. Here is what I did instead, in case it helps anyone:

```sh
ssh my-birdnet-pi
cd BirdNET-Pi
./scripts/stop_core_services.sh
```

Manually back up these files:

```
BirdNET-Pi/BirdDB.txt
BirdSongs/
BirdNET-Pi/apprise.txt
BirdNET-Pi/birdnet.conf
BirdNET-Pi/birds.db
BirdNET-Pi/disk_check_exclude.txt
BirdNET-Pi/exclude_species_list.txt
BirdNET-Pi/include_species_list.txt
```

Follow @blogdemoi's [steps](https://github.com/Nachtzuster/BirdNET-Pi/discussions/246#discussioncomment-11423209):

```sh
ssh my-birdnet-pi
cd BirdNET-Pi
git remote remove origin
git remote add origin https://github.com/Nachtzuster/BirdNET-Pi.git
./scripts/update_birdnet.sh
```

**Comment by Nachtzuster** [comment]
@slnc  [migrating](https://github.com/Nachtzuster/BirdNET-Pi#migrating) :smiley:
